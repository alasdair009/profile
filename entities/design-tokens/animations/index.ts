export { curves } from "./curves";
export { prefersReducedMotion } from "./motion";
