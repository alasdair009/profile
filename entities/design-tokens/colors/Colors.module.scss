@use "../../../app/styles/colors";

:export {
  colorblackEvil: colors.$colorblackEvil;
  colorGreyLight: colors.$colorGreyLight;
  colorGreyDark: colors.$colorGreyDark;
  colorWhiteGhost: colors.$colorWhiteGhost;

  colorBlueSea: colors.$colorBlueSea;
  colorGreenGrass: colors.$colorGreenGrass;
  colorRedHeat: colors.$colorRedHeat;
}
