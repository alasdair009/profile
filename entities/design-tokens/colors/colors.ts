export const colorsGrey = {
  blackEvil: "#000",
  greyDark: "#222425",
  whiteGhost: "#fff",
};

export const colorsBrand = {
  blueSea: "#3487F5",
  greenGrass: "#2E9B26",
};

export const colors = {
  ...colorsGrey,
  ...colorsBrand,
};
