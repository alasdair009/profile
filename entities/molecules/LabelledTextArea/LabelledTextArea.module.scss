@use "../../../app/styles/dimensions";
@use "../../../app/styles/input";
@use "../../../app/styles/helpers/rem";

.root {
  @include input.labelledStyles;
}

.labelText {
  display: block;
  max-width: rem.toRem(dimensions.$globalInputMaxWidth);
  text-align: center;
  width: 100%;
}
