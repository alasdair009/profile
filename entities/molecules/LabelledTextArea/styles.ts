"use client";
import styled from "styled-components";
import { rem } from "polished";
import { globalInputMaxWidth, sizes } from "../../design-tokens/dimensions";

export const Root = styled.label`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto ${sizes.s8.rem};
  max-width: ${rem(globalInputMaxWidth)};
  width: 100%;
`;

export const LabelText = styled.span`
  display: block;
  max-width: ${rem(globalInputMaxWidth)};
  text-align: center;
  width: 100%;
`;
