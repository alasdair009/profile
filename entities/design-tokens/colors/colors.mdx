import { Meta, ColorPalette, ColorItem } from "@storybook/addon-docs/blocks";
import { colorsBrand, colorsGrey } from "./colors";
import { CodeBlock } from "../../molecules/CodeBlock";

<Meta title="Design Tokens/Colors" />

# Colors

Colours used throughout the site are available as SCSS variables and Javascript objects.

<ColorPalette>
  <ColorItem
    title="Greys"
    subtitle="Grey colours used mostly for background and copy."
    colors={colorsGrey}
  />
  <ColorItem
    title="Brand"
    subtitle="Main brand decorative colours."
    colors={colorsBrand}
  />
</ColorPalette>

<CodeBlock
  language="scss"
  code={`@use "styles/colors";\n\n.element{\n   background: colors.$colorGreenGrass\n}`}
  description="A colour used in scss"
/>

<CodeBlock
  language="typescript"
  code={`import { colors } from "@entities";\n\nconst style: CSSProperties = {\n   background: colors.greenGrass,\n};`}
  description="A colour used in ts"
/>
