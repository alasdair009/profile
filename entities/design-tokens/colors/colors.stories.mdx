import { Meta, ColorPalette, ColorItem } from "@storybook/blocks";
import {colorsBrand, colorsGrey} from "./colors";

<Meta title="Design Tokens/Colors" />

# Colors

<ColorPalette>
    <ColorItem
        title="Greys"
        subtitle="Some of the greys"
        colors={colorsGrey}
    />
    <ColorItem
        title="Brand"
        subtitle="Main brand colors"
        colors={colorsBrand}
    />
</ColorPalette>