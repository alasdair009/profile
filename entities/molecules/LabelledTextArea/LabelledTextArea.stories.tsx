import { LabelledTextArea } from "./LabelledTextArea";
import { Meta, StoryObj } from "@storybook/react";
import { expect, userEvent, within } from "@storybook/test";
import { colors } from "../../design-tokens/colors";

const meta: Meta<typeof LabelledTextArea> = {
  component: LabelledTextArea,
  args: {
    label: "Text area input",
    required: true,
    isInvalid: false,
    defaultValue: "hello world!",
    name: "textareainput",
  },
};
export default meta;

export const Default: StoryObj<typeof LabelledTextArea> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const labelledTextAreaElement = canvas.getAllByTestId(
      LabelledTextArea.name
    )[0];
    const baseTextAreaElement: HTMLTextAreaElement = canvas.getByLabelText(
      args.label
    );
    const labelElement = canvas.getByText(args.label);

    await expect(labelledTextAreaElement).toBeInTheDocument();
    await expect(baseTextAreaElement).toBeInTheDocument();
    await expect(labelElement).toBeInTheDocument();

    await userEvent.clear(baseTextAreaElement);
    await userEvent.type(baseTextAreaElement, "bye world");
  },
};

export const Invalid: StoryObj<typeof LabelledTextArea> = {
  args: {
    required: true,
    isInvalid: true,
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const baseTextAreaElement = canvas.getByLabelText(args.label);

    await expect(baseTextAreaElement).toBeInTheDocument();
    await expect(baseTextAreaElement).toHaveStyle(
      "background-color: rgba(223,28,65,0.15)"
    );

    await userEvent.type(baseTextAreaElement, "bye world");

    await expect(baseTextAreaElement).toHaveStyle(
      `background-color: ${colors.greyDark}`
    );
  },
};
