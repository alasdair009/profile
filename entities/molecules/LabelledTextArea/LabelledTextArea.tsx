import { Root } from "./styles";
import { HTMLAttributes } from "react";
import { BaseTextArea } from "../../atoms/BaseTextArea";
import { LabelText } from "./styles";

type LabelledTextAreaProps = {
  /**
   * Label text for the field.
   */
  label: string;
  /**
   * Value is required for valid entry.
   */
  required?: boolean;
  /**
   * Field displays as invalid entry.
   */
  isInvalid?: boolean;
  /**
   * Default value of the field.
   */
  value?: string;
  /**
   * Name of the field.
   */
  name: string;
} & HTMLAttributes<HTMLTextAreaElement>;

/**
 * HTML textarea input field.
 */
export function LabelledTextArea({
  label,
  name,
  ...rest
}: LabelledTextAreaProps) {
  return (
    <Root data-testid={LabelledTextArea.name}>
      <LabelText>{label}</LabelText>
      <BaseTextArea {...rest} />
    </Root>
  );
}
