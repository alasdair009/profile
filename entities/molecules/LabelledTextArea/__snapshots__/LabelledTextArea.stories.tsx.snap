// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/LabelledTextArea Default play-test 1`] = `
<label data-testid="LabelledTextArea"
       class="sc-braxZu jKGkOx"
>
  <span class="sc-gJhJTp oKJJG">
    Text area input
  </span>
  <textarea required
            class="sc-elDIKY jEuff"
  >
    hello world!
  </textarea>
</label>
`;

exports[`molecules/LabelledTextArea Invalid play-test 1`] = `
<label data-testid="LabelledTextArea"
       class="sc-braxZu jKGkOx"
>
  <span class="sc-gJhJTp oKJJG">
    Text area input
  </span>
  <textarea required
            class="sc-elDIKY jEuff"
  >
    hello world!
  </textarea>
</label>
`;
